appSyncClient.go

```diff
@@ -118,38 +181,86 @@ func (client *appSyncClient) CreateOrUpdateResolvers(apiID string, resolversFile
+
+		shouldContinue := true
+		if len(resolver.Functions) > 0 {
+			resolverKind = appsync.ResolverKindPipeline
+			pipelineConfig = &appsync.PipelineConfig{}
+
+			for i := range resolver.Functions {
+				existingFunction := findFunctionByName(resolver.Functions[i], functions)
+
+				if existingFunction == nil {
+					logger.Printf("Failed to find function: %s, I will not continue with updating resolver type: %s, field: %s\n",
+						resolver.Functions[i], resolver.TypeName, resolver.FieldName)
+					shouldContinue = false
+					break
+				}
+
+				pipelineConfig.Functions = append(pipelineConfig.Functions, existingFunction.FunctionId)
+			}
+
+			dataSourceName = nil
+		}
+
+		if !shouldContinue {
+			continue
+		}
+
 		if resolverResp != nil {
 			var params = &appsync.UpdateResolverInput{
 				ApiId:                   aws.String(apiID),
-				DataSourceName:          aws.String(resolver.DataSourceName),
+				DataSourceName:          dataSourceName,
 				FieldName:               aws.String(resolver.FieldName),
-				RequestMappingTemplate:  aws.String(fmt.Sprintf("%s", resolver.RequestMappingTemplate)),
+				RequestMappingTemplate:  aws.String(resolver.RequestMappingTemplate),
 				ResponseMappingTemplate: aws.String(resolver.ResponseMappingTemplate),
 				TypeName:                aws.String(resolver.TypeName),
+				Kind:                    &resolverKind,
+				PipelineConfig:          pipelineConfig,
 			}
 			_, err := client.updateResolver(params)
 			if err != nil {
-				nResolversfailUpdate++
+				fmt.Printf("Resolver on type %s and field %s failed to update: %s", resolver.TypeName, resolver.FieldName, err)
+				resolverStatistics.FailedToUpdate++
+			} else {
+				resolverStatistics.Updated++
 			}
-			nResolversSuccessfullyUpdated++
 		} else {
 			var params = &appsync.CreateResolverInput{
 				ApiId:                   aws.String(apiID),
-				DataSourceName:          aws.String(resolver.DataSourceName),
+				DataSourceName:          dataSourceName,
 				FieldName:               aws.String(resolver.FieldName),
-				RequestMappingTemplate:  aws.String(fmt.Sprintf("%s", resolver.RequestMappingTemplate)),
+				RequestMappingTemplate:  aws.String(resolver.RequestMappingTemplate),
 				ResponseMappingTemplate: aws.String(resolver.ResponseMappingTemplate),
 				TypeName:                aws.String(resolver.TypeName),
+				Kind:                    &resolverKind,
+				PipelineConfig:          pipelineConfig,
 			}
 			_, err := client.createResolver(params)
 			if err != nil {
-				nResolversfailCreated++
+				fmt.Printf("Resolver on type %s and field %s failed to create: %s", resolver.TypeName, resolver.FieldName, err)
+				resolverStatistics.FailedToCreate++
+			} else {
+				resolverStatistics.Created++
 			}
+		}
+	}
 
-			nResolversSuccessfullyCreated++
+	return resolverStatistics, functionStatistics, nil
+}
+
+func findFunctionByName(name string, functions []*appsync.FunctionConfiguration) *appsync.FunctionConfiguration {
```

Comment:
Should we call this getFunctionByName to match the naming convention in this package